### PR TITLE
[DRAFT]add azure maps account feature

### PR DESCRIPTION
**Pull request by reshmarajan1730**
# [Issue-id](https://github.com/aztfmod/terraform-azurerm-caf/issues/1713)

## PR Checklist

---

- [x] I have added example(s) inside the [./examples/] folder
- [ ] I have added the example(s) to the integration test list for [normal (~30 minutes)](./workflows/standalone-scenarios.json) or [long runner >30 minutes](./workflows/standalone-scenarios-longrunners.json)
- [x] I have checked the [coding conventions as per the wiki](https://github.com/aztfmod/terraform-azurerm-caf/wiki)
- [x] I have checked to ensure there aren't other open Pull Requests for the same update/change?

## Does this introduce a breaking change

- [ ] YES
- [x] NO

## Testing
This has been tested (standalone) in inhouse azure account

### CODE REVIEW

File: examples/maps_account/101-azure-maps-account/configuration.tfvars [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  }
+}
+
+resource_groups = {
+  test = {
+    name = "rg-maps-test"
+  }
+}
+
+maps_account = {
```
Comment (LaurentLesle) [inline examples/maps_account/101-azure-maps-account/configuration.tfvars:14]:
```suggestion
maps_accounts = {
```

Suggest you use the plural version for top variables

File: examples/module.tf [hunk 2]
```diff
@@ -394,6 +394,9 @@ module "example" {
   purview = {
     purview_accounts = var.purview_accounts
   }
+  maps = {
+    maps_account = var.maps_account
```
Comment (LaurentLesle) [inline examples/module.tf:398]:
```suggestion
    maps_accounts = var.maps_accounts
```

File: examples/variables.tf [hunk 3]
```diff
@@ -235,6 +235,9 @@ variable "storage_accounts" {
 variable "storage_account_file_shares" {
   default = {}
 }
+variable "maps_account" {
```
Comment (LaurentLesle) [inline examples/variables.tf:238]:
```suggestion
variable "maps_accounts" {
```

File: locals.combined_objects.tf [hunk 4]
```diff
@@ -107,6 +107,7 @@ locals {
   combined_objects_logic_app_standard                             = merge(tomap({ (local.client_config.landingzone_key) = module.logic_app_standard }), try(var.remote_objects.logic_app_standard, {}))
   combined_objects_logic_app_workflow                             = merge(tomap({ (local.client_config.landingzone_key) = module.logic_app_workflow }), try(var.remote_objects.logic_app_workflow, {}), try(var.data_sources.logic_app_workflow, {}))
   combined_objects_machine_learning                               = merge(tomap({ (local.client_config.landingzone_key) = module.machine_learning_workspaces }), try(var.remote_objects.machine_learning_workspaces, {}), try(var.data_sources.machine_learning_workspaces, {}))
+  combined_objects_maps_accounts                                  = merge(tomap({ (local.client_config.landingzone_key) = module.maps_account }), try(var.remote_objects.maps_account, {}))
```
Comment (LaurentLesle) [inline locals.combined_objects.tf:110]:
```suggestion
  combined_objects_maps_accounts                                  = merge(tomap({ (local.client_config.landingzone_key) = module.maps_accounts }), try(var.remote_objects.maps_accounts, {}))
```

File: locals.tf [hunk 5]
```diff
@@ -241,6 +241,9 @@ locals {
   cognitive_services = {
     cognitive_services_account = try(var.cognitive_services.cognitive_services_account, {})
   }
+  maps = {
+    maps_account = try(var.maps.maps_account, {})
```
Comment (LaurentLesle) [inline locals.tf:245]:
```suggestion
    maps_accounts = try(var.maps.maps_accounts, {})
```

File: maps_account.tf [hunk 6]
```diff
@@ -0,0 +1,13 @@
+module "maps_account" {
```
Comment (LaurentLesle) [inline maps_account.tf:1]:
```suggestion
module "maps_accounts" {
```

File: maps_account.tf [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  source   = "./modules/maps/maps_account"
+  for_each = try(local.maps.maps_account, {})
```
Comment (LaurentLesle) [inline maps_account.tf:3]:
```suggestion
  for_each = try(local.maps.maps_accounts, {})
```

File: maps_account.tf [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+  global_settings     = local.global_settings
+  resource_group_name = can(each.value.resource_group.name) || can(each.value.resource_group_name) ? try(each.value.resource_group.name, each.value.resource_group_name) : local.combined_objects_resource_groups[try(each.value.resource_group.lz_key, local.client_config.landingzone_key)][try(each.value.resource_group_key, each.value.resource_group.key)].name
+  sku_name            = each.value.sku_name
+  settings            = each.value
+}
+
+output "maps_account" {
```
Comment (LaurentLesle) [inline maps_account.tf:11]:
```suggestion
output "maps_accounts" {
```

File: maps_account.tf [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+  value     = module.maps_account
```
Comment (LaurentLesle) [inline maps_account.tf:12]:
```suggestion
  value     = module.maps_accounts
```

**Review by LaurentLesle — CHANGES_REQUESTED** [review]
I propose to be consistent with the rest of the framework to use plural version for aztfmod top variables

File: maps_account.tf (hunk 9 above)
Comment (reshmarajan1730) [reply maps_account.tf:12]: @LaurentLesle Thanks for the suggestions, I have to close this PR due to some conflicts downstream. I have opened another PR , i will incorporate the changes you suggested into the fixed one.

**Pull request closed without merging**